webapp/src/components/common/dropdown/dropdown.tsx

```diff
@@ -0,0 +1,43 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React, {useCallback, useMemo} from 'react';
+import Select, {MultiValue, SingleValue} from 'react-select';
+
+import Control from 'components/common/dropdown/control';
+
+export type DropdownOption = {
+    value: string,
+    label: string,
+};
+
+export type Props = {
+    options: Array<DropdownOption>
+    value?: DropdownOption
+    defaultValue?: DropdownOption
+    onChange: (newValue: DropdownOption) => void
+}
+const Dropdown = ({options, value, defaultValue, onChange}: Props) => {
+    const customComponents = useMemo(() => {
+        return {
+            Control,
+        };
+    }, []);
+
+    // This handler only serves the purpose of satisfying typescript.
+    // Otherwise it complains abut incorrect type's callback passed to Select's onChange prop.
```

Comment:
We control the type of onChange, right? Can nothing be done on that side so this is more elegant?

Reply:
If this should handle selecting multiple objects, I'd suggest making the this component's `onChange` always expect an array so that the caller can consistently handle that instead of having to figure out the type. This component could then handle wrapping a single result in an array

Reply:
our onChange is very simple and its abstracted from the fact that this component uses react-select underneath it. Its the react-select's onChange thats complicated and I don't want to pass that complication to users of our dropdown component, hence this wrapper.

Reply:
@hmhealey I'll do that when I add the multiselector for team include/exclude setting but yeah, it will become an array.